src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceSinkConfig.java

```diff
@@ -148,6 +162,19 @@ public String getExternalIdField() {
     return externalIdField;
   }
 
+  public String getConcurrencyMode() {
+    return concurrencyMode;
+  }
+
+  public ConcurrencyMode getConcurrencyModeEnum() {
+    try {
+      return ConcurrencyMode.valueOf(concurrencyMode);
```

Comment:
Do we need a null check and default value setting here? Test how pipeline upgrade works when we are upgrading from 1.4.3 to 1.4.4. version of the plugin.

Reply:
Returning the default value: `ConcurrencyMode.Parallel` if 

- the value is null or
- this enum type has no constant with the specified name

Tested with pipeline upgrade and it returns 200 OK. No issues were observed.

e130c2d587656a458efc1c7b64df2f795710543c

Reply:
Please add a log in the exception block, so we know that there was an issue with the concurrency mode.

Reply:
Added logger in the exception catch block. b1f4d0dd2ca351b7b9e6dbc0ad8b5b518a9402c2

Reply:
Please change the log to info, and also mention that parallel mode is being used.

Reply:
Add to the log message that default concurrency mode is used instead. The intention here is to inform that we encountered an issue and took an action.

Reply:
Updated the log message.